ts/components/messages/MessageDueDateBar.tsx

```diff
@@ -57,182 +70,210 @@ const styles = StyleSheet.create({
+    <ExpiredNotExpirableTextContent />
+  ) : status === "valid" ? (
+    <ValidTextContent time={time} date={date} />
+  ) : (
+    assertNever(status)
+  );
+};
 
-  get bannerStyle(): ViewStyle {
-    if (this.paid) {
-      return { backgroundColor: customVariables.brandGray };
-    }
-
-    if (this.isPaymentExpired) {
-      return {
-        backgroundColor: this.isPaymentExpirable
-          ? customVariables.brandDarkGray
-          : customVariables.calendarExpirableColor
-      };
-    }
-
-    if (this.isPaymentExpiring) {
-      return { backgroundColor: customVariables.calendarExpirableColor };
-    }
-    return { backgroundColor: customVariables.brandGray };
-  }
+const getCalendarIconBackgoundColor = (status: PaymentStatus) =>
+  status === "paid"
+    ? customVariables.lighterGray
+    : status === "expiring" ||
+      status === "expiredAndExpirable" ||
+      status === "expiredNotExpirable"
+    ? customVariables.colorWhite
+    : status === "valid"
+    ? customVariables.brandDarkGray
+    : assertNever(status);
 
-  get textContent() {
-    const { dueDate: maybeDueDate } = this;
-    if (maybeDueDate.isNone()) {
-      return undefined;
-    }
-    const dueDate = maybeDueDate.value;
-    const time = format(dueDate, "HH.mm");
-    const date = formatDateAsLocal(dueDate, true, true);
-
-    if (this.paid) {
-      return (
-        <React.Fragment>
-          {`${I18n.t("messages.cta.payment.paid")} `}
-          <Text link={true} onPress={this.props.onGoToWallet}>
-            {I18n.t("wallet.wallet")}
-          </Text>
-        </React.Fragment>
-      );
-    }
-
-    if (this.isPaymentExpiring) {
-      return (
-        <React.Fragment>
-          {I18n.t("messages.cta.payment.expiringAlert.block1")}
-          <Text bold={true} white={true}>{` ${date} `}</Text>
-          {I18n.t("messages.cta.payment.expiringAlert.block2")}
-          <Text bold={true} white={true}>{` ${time} `}</Text>
-        </React.Fragment>
-      );
-    }
-
-    if (this.isPaymentExpired) {
-      if (this.isPaymentExpirable) {
-        return (
-          <React.Fragment>
-            {I18n.t("messages.cta.payment.expiredAlert.expirable.block1")}
-            <Text bold={true} white={true}>{` ${time} `}</Text>
-            {I18n.t("messages.cta.payment.expiredAlert.expirable.block2")}
-            <Text bold={true} white={true}>{` ${date}`}</Text>
-          </React.Fragment>
-        );
-      }
-      return (
-        <React.Fragment>
-          {I18n.t("messages.cta.payment.expiredAlert.unexpirable.block")}
-        </React.Fragment>
-      );
-    }
-
-    return (
-      <React.Fragment>
-        {I18n.t("messages.cta.payment.addMemo.block1")}
-        <Text bold={true}>{` ${date} `}</Text>
-        {"["}
-        {I18n.t("messages.cta.payment.addMemo.block2")}
-        <Text bold={true}>{` ${time}`}</Text>
-        {"]"}
-      </React.Fragment>
-    );
-  }
+const getCalendarTextColor = (status: PaymentStatus) =>
```

Comment:
Same as above, you could use a switch instead of multiple ternary statements.
Also, this block could be defined outside the component: it this way this code is created each time the component is renderered.
Another approach is using mapping (But it is still more correct to use the switch (this is because if you add another state the switch ensure you'll implement the new case))

```typescript
const backgroundColorMap: Map<PaymentStatus, string> = new Map<
  PaymentStatus,
  string
>([
  ["paid", customVariables.lighterGray],
  ["expiring", customVariables.colorWhite],
  ["expiredNotValid", customVariables.colorWhite],
  ["expiredButValid", customVariables.colorWhite],
  ["valid", customVariables.brandDarkGray]
]);
const getCalendarIconBackgoundColor = (status: PaymentStatus) =>
  fromNullable(backgroundColorMap.get(status)).getOrElse(
    customVariables.brandDarkGray
  );

```

Reply:
> it this way this code is created each time the component is renderered.

I didn't get this

Reply:
Since `getCalendarTextColor` is defined inside the component when a prop or state changes React renders that component from scratch within all its functions. So that function will be created on each render.
`getCalendarTextColor` is a function that could be created once. So my suggestion is put it out of the component body so it will be defined once.

React component are functions and functions defined in a function have their own scope

Reply:
I think `getCalendarTextColor` is already outside the component, isn't it?